base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class Decorator < Nameable
+  def initialize
+    super()
+    @nameable = Nameable.new
+  end
+
+  def correct_name
+    @nameable.correct_name(@name)
```

Comment:
- [ ] You don't need to pass the @name instance variable. Let's go like this: `@nameable.correct_name`